src/Nethermind/Nethermind.Synchronization/FastBlocks/FastHeadersSyncFeed.cs

```diff
@@ -201,8 +201,25 @@ protected virtual void ResetPivot()
             BlockHeader? lowestInserted = _blockTree.LowestInsertedHeader;
             if (lowestInserted is not null && lowestInserted!.Number < _pivotNumber)
             {
-                SetExpectedNextHeaderToParent(lowestInserted);
-                _lowestRequestedHeaderNumber = lowestInserted.Number;
+                if (lowestInserted.TotalDifficulty is null)
```

Comment:
So this is the crux, that the `lowestInserted` can be loaded without the `TotalDifficulty`. Why do we load it though this way. Would it not be easier to handle it where it's loaded and pass requirements https://github.com/NethermindEth/nethermind/blob/4fa9b3ffeeafd6f4715780348f02da37ec4bdc5a/src/Nethermind/Nethermind.Blockchain/BlockTree.Initializer.cs#L133 so that the `lowestInserted` is properly preloaded??

Reply:
Heck, I don't know. Strictly speaking, only the hash is stored.